test/unit/Site.test.js

```diff
@@ -0,0 +1,323 @@
+  expect(fs.readFileSync(path.resolve('newDir/_markbind/variables.md'), 'utf8'))
+    .toEqual(USER_VARIABLES_DEFAULT);
+
+  // site.json
+  expect(fs.existsSync(path.resolve('newDir/site.json'))).toEqual(true);
+  expect(fs.readJsonSync(path.resolve('newDir/site.json'))).toEqual(JSON.parse(SITE_JSON_DEFAULT));
+
+  // index.md
+  expect(fs.existsSync(path.resolve('newDir/index.md'))).toEqual(true);
+  expect(fs.readFileSync(path.resolve('newDir/index.md'), 'utf8')).toEqual(INDEX_MD_DEFAULT);
+});
+
+test('Site baseurls are correct for sub nested subsites', async () => {
+  const json = {
+    'lib/template/page.ejs': PAGE_EJS,
+    'site.json': SITE_JSON_DEFAULT,
+    'sub/site.json': SITE_JSON_DEFAULT,
+    'sub/sub/site.json': SITE_JSON_DEFAULT,
+    'otherSub/sub/site.json': SITE_JSON_DEFAULT,
+  };
+  fs.vol.fromJSON(json, '');
+
+  const baseUrlMapExpected = {};
+  baseUrlMapExpected[path.resolve('')] = true;
+  baseUrlMapExpected[path.resolve('sub')] = true;
+  baseUrlMapExpected[path.resolve('sub/sub')] = true;
+  baseUrlMapExpected[path.resolve('otherSub/sub')] = true;
+
+  const site = new Site('./', '_site');
+  await site.collectBaseUrl();
+  expect(site.baseUrlMap).toEqual(baseUrlMapExpected);
+});
+
+test('Site removeAsync removes the correct asset', async () => {
+  const json = {
+    'lib/template/page.ejs': PAGE_EJS,
+    '_site/toRemove.jpg': '',
+    '_site/dontRemove.png': '',
+    'toRemove.html': '',
+  };
+  fs.vol.fromJSON(json, '');
+
+  const site = new Site('./', '_site');
+  await site.removeAsset('toRemove.jpg');
+  expect(fs.existsSync(path.resolve('_site/toRemove.jpg'))).toEqual(false);
+  expect(fs.existsSync(path.resolve('_site/dontRemove.png'))).toEqual(true);
+});
+
+test('Site baseurls are correct for sub nested subsites', async () => {
+  const json = {
+    'lib/template/page.ejs': PAGE_EJS,
+    'site.json': SITE_JSON_DEFAULT,
+    'sub/site.json': SITE_JSON_DEFAULT,
+    'sub/sub/site.json': SITE_JSON_DEFAULT,
+    'otherSub/sub/site.json': SITE_JSON_DEFAULT,
+  };
+  fs.vol.fromJSON(json, '');
+
+  const baseUrlMapExpected = {};
+  baseUrlMapExpected[path.resolve('')] = true;
+  baseUrlMapExpected[path.resolve('sub')] = true;
+  baseUrlMapExpected[path.resolve('sub/sub')] = true;
+  baseUrlMapExpected[path.resolve('otherSub/sub')] = true;
+
+  const site = new Site('./', '_site');
+  await site.collectBaseUrl();
+  expect(site.baseUrlMap).toEqual(baseUrlMapExpected);
+});
+
+test('Site read site config for default', async () => {
+  const json = {
+    'lib/template/page.ejs': PAGE_EJS,
+    'site.json': SITE_JSON_DEFAULT,
+  };
+  fs.vol.fromJSON(json, '');
+
+  const site = new Site('./', '_site');
+  await site.readSiteConfig();
+  expect(site.siteConfig).toEqual(JSON.parse(SITE_JSON_DEFAULT));
+});
+
+test('Site read site config for custom site config', async () => {
+  const customSiteJson = {
+    baseUrl: '',
+    pages: [
```

Comment:
This `site.json` will have to be changed to have the new attribute `addressable` once https://github.com/MarkBind/markbind/pull/175 is merged.